### PR TITLE
feat: hide social banner on scroll

### PR DESCRIPTION
**Pull request by akulsr0**
This hides social banner when scrolled on mobile/tablet.

**Demo**
ezgif com-gif-maker

### CODE REVIEW

**Comment by akulsr0** [comment]
@harish-sethuraman @gaearon any feedback/suggestions here?

**Comment by harish-sethuraman** [comment]
Hey 👋🏻  There is one another PR that fixes the same thing 
- https://github.com/reactjs/reactjs.org/pull/4713

**Comment by akulsr0** [comment]
@harish-sethuraman Yeah, I see there were some changes requested and that pr was not active ~1 month, so I tried a different approach here to fix the issue.
Since on smaller devices, the social banner is taking much space, it would be good to fix this with either approach :)

**Comment by akulsr0** [comment]
Bump?

File: beta/src/components/Layout/Sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -13,32 +13,44 @@ import {MobileNav} from '../Nav/MobileNav';
...
 export function Sidebar() {
   const {menuRef, isOpen} = React.useContext(MenuContext);
   const isMobileSidebar = useMediaQuery(SIDEBAR_BREAKPOINT);
   let routeTree = React.useContext(SidebarContext);
   const isHidden = isMobileSidebar ? !isOpen : false;
 
+  const bannerHeight = useMediaQuery(SM_BREAKPOINT) ? 72 : 40;
+  let navMarginTop = bannerHeight;
+  if (typeof window !== 'undefined') {
+    const scrollAmount = document.documentElement.scrollTop;
```
Comment (gaearon) [inline beta/src/components/Layout/Sidebar/Sidebar.tsx:27]: We shouldn't be reading global mutable values during render. This usually indicates a bug (even when it's not obvious how to trigger it).

File: beta/src/components/Layout/Sidebar/Sidebar.tsx [hunk 2]
```diff
@@ -13,32 +13,44 @@ import {MobileNav} from '../Nav/MobileNav';
...
 
 export function Sidebar() {
   const {menuRef, isOpen} = React.useContext(MenuContext);
   const isMobileSidebar = useMediaQuery(SIDEBAR_BREAKPOINT);
   let routeTree = React.useContext(SidebarContext);
   const isHidden = isMobileSidebar ? !isOpen : false;
 
+  const bannerHeight = useMediaQuery(SM_BREAKPOINT) ? 72 : 40;
+  let navMarginTop = bannerHeight;
+  if (typeof window !== 'undefined') {
```
Comment (gaearon) [inline beta/src/components/Layout/Sidebar/Sidebar.tsx:26]: Different render outputs for server and initial client output means hydration can produce mismatches. This doesn't seem right.

**Review by gaearon — CHANGES_REQUESTED** [review]
From the coding perspective, this solution doesn't seem to follow React's rules. There should be the same render output for server and initial client, and render shouldn't read mutable values. So I don't think we can merge this.

**Pull request closed without merging**